### PR TITLE
[user-authz] restrict access to machine resources 

### PR DESCRIPTION
**Pull request by borg-z**
## Description

Added additional RBAC rules and ValidatingAdmissionPolicy to restrict actions on `cluster.x-k8s.io/machine.sapcloud.io` resources (machines, machinesets, machinedeployments). ValidatingAdmissionPolicy is needed to prohibit deletion of machine resources for all users, even super administrators, except for d8 service accounts.

## Why do we need it, and what problem does it solve?

The ability to delete or modify machine resources should be restricted to cluster administrators only. This ensures that only authorized personnel can perform critical operations on machine resources.

## What is the expected result?

Only users with ClusterAdmin privileges will be able to change machine resources. Only d8 service accounts can delete  machine resources.

## Checklist
- [ ] The code is covered by unit tests.
- [x] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [x] Changes were tested in the Kubernetes cluster manually.

## Changelog entries

```changes
section: node-manager
type: fix
summary: Restricted actions on `cluster.x-k8s.io/machine.sapcloud.io`
impact: Unauthorized users will be unable to manage `cluster.x-k8s.io/machine.sapcloud.io` resources (`machines`, `machinesets`, `machinedeployments`).
impact_level: default
```

### CODE REVIEW

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/vcd'...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 9941986940 9941986940-1-con-1-27 /sys/deckhouse-oss/install:pr9026 redos@62.84.119.178

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/yandex-cloud'...

File: modules/002-deckhouse/templates/validation.yaml [hunk 1]
```diff
@@ -99,4 +99,47 @@ spec:
...
+        apiVersions: ["*"]
+        operations: ["DELETE"]
+        resources: ["*"]
+  matchConditions:
+    - name: 'allow-only-for-d8-accounts'
+      expression: '!(request.userInfo.username.startsWith("system:serviceaccount:d8-") || request.userInfo.username == "kubernetes-admin")'
+  validations:
+    - expression: 'false'
+      reason: Forbidden
+      messageExpression: '''Operation on machine object is forbidden for this user by ValidatingAdmissionPolicy {{ $policyName }}'''
```
Comment (miklezzzz) [inline modules/002-deckhouse/templates/validation.yaml:129]: what is the target milestone? As I remember, matchConditions option isn't supported in 1.26

**Review by miklezzzz — APPROVED** [review]
what is the milestone? as I remember, matchCondtion

File: modules/002-deckhouse/templates/validation.yaml (hunk 1 above)
Comment (borg-z) [reply modules/002-deckhouse/templates/validation.yaml]: fixed

**Comment by borg-z** [comment]
Test result:

```
root@yc-9026-master-0:~# kubectl config view --minify -o jsonpath='{.users[0].name}' && echo
kubernetes-admin
root@yc-9026-master-0:~# kubectl -n d8-cloud-instance-manager delete machines.machine.sapcloud.io zykov-yc-9026-worker-3c277896-5cf65-2h4cs
machine.machine.sapcloud.io "zykov-yc-9026-worker-3c277896-5cf65-2h4cs" deleted
```

```
➜  ~ kubectl get ClusterAuthorizationRule admin -o json | jq .spec.accessLevel
"SuperAdmin"
➜  ~ kubectl -n d8-cloud-instance-manager delete machines.machine.sapcloud.io zykov-yc-9026-worker-3c277896-5cf65-2h4cs
Error from server (Forbidden): machines.machine.sapcloud.io "yc-9026-worker-3c277896-5cf65-2h4cs" is forbidden: ValidatingAdmissionPolicy 'machines.deckhouse.io' with binding 'machines.deckhouse.io-binding' denied request: Operation on machine object is forbidden for this user by ValidatingAdmissionPolicy machines.deckhouse.io
```

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've started the workflow for label 'e2e/run/gcp'...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10158495430 10158495430-2-con-1-27 /sys/deckhouse-oss/install:pr9026 redos@51.250.68.143

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10158495430 10158495430-3-con-1-27 /sys/deckhouse-oss/install:pr9026 redos@51.250.76.137

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by z9r5** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10158495430 10158495430-1-con-1-27 /sys/deckhouse-oss/install:pr9026 redos@89.169.138.0

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @z9r5. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by name212** [comment]
/e2e/abort yandex-cloud;WithoutNAT;containerd;1.27 10333413105 10333413105-1-con-1-27 /sys/deckhouse-oss/install:pr9026 redos@51.250.8.135

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @name212. I've recognized your '/e2e/abort' command and started the workflow...

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @borg-z. I've started the workflow for label 'e2e/run/openstack'...

**Review by name212 — APPROVED** [review]

**Pull request merged**